### PR TITLE
Hallucination Refactor

### PR DESCRIPTION
**Pull request by Tails2091**
General cleanup.
Replaced all but one spawn() with an addtimer(), formatted all for loops, and changed booleans to use TRUE and FALSE instead of 0 and 1.

:cl:
fix: Refactored Hallucinations to match contribution guide standards a bit more.
/:cl:

### CODE REVIEW

**Comment by Tails2091** [comment]
@Birdtalon Just formatting to fit the contribution guide.

**Comment by Birdtalon** [comment]
Fair point then 👍

**Comment by Tails2091** [comment]
@Crazylemon64 Added changelog here and to Claw Game PR.

File: code/modules/flufftext/Hallucination.dm [hunk 1]
```diff
@@ -605,27 +599,26 @@ Gunshots/explosions/opening doors/less rare audio (done)
 /obj/effect/fake_attacker/New(loc, mob/living/carbon/T)
 	..()
 	my_target = T
-	spawn(300)
-		qdel(src)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 300)
 	step_away(src,my_target,2)
-	spawn(0)
-		attack_loop()
```
Comment (variableundefined) [inline code/modules/flufftext/Hallucination.dm]: Never use addtimer for spawn(0), use INVOKE_ASYNC(src, .proc/attack_loop) instead

**Pull request merged**